bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TupleValueImpl.java

```diff
@@ -740,6 +760,12 @@ protected void unshift(long index, Object[] vals) {
 
     // private methods
 
+    private void prepareForAddForcefully(int index, int currentArraySize) {
+        ensureCapacity(index + 1, currentArraySize);
+        fillValues(index);
+        resetSize(index);
+    }
+
```

Comment:
Can we check the possibility to move this to the `AbstractArrayValue` class since similar found in ArrayValueImpl as well?